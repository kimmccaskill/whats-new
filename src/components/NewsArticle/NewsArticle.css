.article-card {
  height: 35%;
  width: 25%;
  border: white solid 1px;
  margin: 10px;
  background-color: white;
}

.card-image {
  width: 100%;
  height: 200px;
}
