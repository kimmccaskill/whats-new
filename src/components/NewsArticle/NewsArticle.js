import React from 'react';
import './NewsArticle.css';
// Each NewsArticle card will display: the image, the headline, the description, and the url (see the screenshot above for an idea).

function NewsArticle(props) {
  return (
    <div className='article-card'>
      <img className="card-image" src={props.image} alt={props.headline} />
      <div>
        <h3>{props.headline}</h3>
        <p>{props.description}</p>
      </div>
    </div>
  )
}

export default NewsArticle;
