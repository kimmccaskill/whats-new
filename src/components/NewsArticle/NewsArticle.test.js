import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';
import local from '../../data/local';

describe('NewsArticle', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<NewsArticle
      key='123'
      image='https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376'
      headline="Jimmy trapped in well"
      description="Boy named Jimmy well down well.  Found by dog."
      url='https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/'
    />);

    expect(wrapper).toMatchSnapshot();
  });

  
});
