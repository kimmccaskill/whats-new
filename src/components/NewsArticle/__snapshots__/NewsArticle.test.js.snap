// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all data passed in correctly 1`] = `
<div
  className="article-card"
>
  <img
    alt="Jimmy trapped in well"
    className="card-image"
    src="https://i0.wp.com/wp-cpr.s3.amazonaws.com/uploads/2019/09/cpr-swolf_black-cube-monumental_DSC8170.jpg?resize=2064,1376"
  />
  <div>
    <h3>
      Jimmy trapped in well
    </h3>
    <p
      className="card-description"
    >
      Boy named Jimmy well down well.  Found by dog.
    </p>
    <a
      href="https://www.cpr.org/2019/09/27/giant-chicken-skeleton-in-denvers-central-library-stands-for-more-than-just-dinner/"
    >
      Check it out
    </a>
  </div>
</div>
`;
